spraybot_row_detection/spraybot_row_detection/processing.py

```diff
@@ -0,0 +1,140 @@
+from spraybot_row_detection.pose import Pose_2D
+
+
+class RowDetectionProcessing(Node):
+
+    def parameters_callback(self, params):
+        for param in params:
+            if param.name == 'y_thresh':
+                self.y_thresh_ = param.value
+            if param.name == 'x_pred':
+                self.x_pred_ = param.value
+            if param.name == 'dist_threshold':
+                self.dist_threshold_ = param.value
+            if param.name == 'weight_flag':
+                self.weight_flag_ = param.value
+            if param.name == 'desired_x':
+                self.desired_x = param.value
+            if param.name == 'min_pts_row':
+                self.min_pts_row_ = param.value
+            if param.name == 'alpha':
+                self.alpha_ = param.value
+        return SetParametersResult(successful=True)
+
+    def __init__(self, node):
+        self.node_ = node
+        self.node_.declare_parameters(
+            namespace='',
+            parameters=[
+                ('y_thresh', 1.5),
+                ('x_pred', 10.0),
+                ('dist_threshold', 4.0),
+                ('weight_flag', False),
+                ('desired_x', 1.0),
+                ('min_pts_row', 1),
+                ('alpha', 0.2),
+            ],
+        )
+        self.y_thresh_ = self.node_.get_parameter('y_thresh').value
+        self.x_pred_ = self.node_.get_parameter('x_pred').value
+        self.dist_threshold_ = self.node_.get_parameter('dist_threshold').value
+        self.weight_flag_ = self.node_.get_parameter('weight_flag').value
+        self.desired_x = self.node_.get_parameter('desired_x').value
+        self.min_pts_row_ = self.node_.get_parameter('min_pts_row').value
+        self.alpha_ = self.node_.get_parameter('alpha').value
+        self.node_.add_on_set_parameters_callback(self.parameters_callback)
+
+    def get_weights(self, x_arr, triangle_start=3.0, k=0.2):
+        wt = []
+        for x in x_arr:
+            if x > triangle_start or x < -triangle_start:
+                wt.append(k * (self.dist_threshold_ - abs(x)))
+            else:
+                wt.append(1.0)
+        return wt
+
+    def group_points(self, msg, selected_pts):
+        has_points = True
+        left_pts = []
+        right_pts = []
+        all_pts = []
+        for point in msg.points:
+            if (
+                point.y < 0
+                and point.y > -self.y_thresh_
+                and abs(point.x) <= self.dist_threshold_
+            ):
+                pt = Point(point.x, point.y, point.z)
+                left_pts.append(pt)
+                selected_pts.add_points(pt)
+            elif (
+                point.y > 0
+                and point.y < self.y_thresh_
+                and abs(point.x) <= self.dist_threshold_
+            ):
+                pt = Point(point.x, point.y, point.z)
+                right_pts.append(pt)
+                selected_pts.add_points(pt)
```

Comment:
This can be simplified:
```suggestion
            if abs(point.y) < self.y_thresh_ and abs(point.x) <= self.dist_threshold_:
                pt = Point(point.x, point.y, point.z)
                selected_pts.add_points(pt)
                if points.y < 0:
                  left_pts.append(pt)
                else:
                  right_pts.append(pt)
```